warehouse/query-core/src/main/java/datawave/query/config/ShardQueryConfiguration.java

```diff
@@ -713,6 +723,69 @@ public ShardQueryConfiguration(ShardQueryConfiguration other) {
         this.setSortQueryByCounts(other.isSortQueryByCounts());
     }
 
+    /**
+     * This constructor is used when we are creating a checkpoint for a set of ranges (i.e. QueryData objects). All configuration required for post planning
+     * needs to be copied over here.
+     *
+     * @param other
+     * @param queries
+     */
+    public ShardQueryConfiguration(ShardQueryConfiguration other, Collection<QueryData> queries) {
```

Comment:
This probably needs to be updated.

Reply:
Looking at this a little closer, I guess it's ok.  The full copy constructor can manage copying all of the fields, while this one only handles the ones that are important for a checkpoint.  @ivakegg how did you decide on the list of members that need to be copied to support checkpointable queries, and could these members be moved to the CheckpointableQueryConfiguration class (or something equivalent) to better distinguish between properties that are important to checkpointable queries, and properties that aren't?

Reply:
The common fields required are already covered by the GenericQueryConfiguration copy constructor.  I am cleaning out any redundencies however.

Reply:
Found some redundency with the super copy constructor and some incorrectly overridden Query members.